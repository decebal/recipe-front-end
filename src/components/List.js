// @flow

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Button, Card, CardBlock, CardDeck, CardTitle,
} from '@bootstrap-styled/v4/dist/@bootstrap-styled/v4';
import type Recipe from '../types/recipe';
import { Headline } from './layout';
import type Recipes from '../types/recipes';

const PaddedCard = styled.div`
  margin: 10px;
`;

const recipesList = ({ recipes }) => Object.values(recipes).map((item: Recipe) => (
  <PaddedCard key={`recipe-${item.id}`}>
    <Card width="15rem">
      <CardBlock>
        <CardTitle>{item.name}</CardTitle>
        <Button color="primary" href={`/recipe/${item.id}`}>See more</Button>
      </CardBlock>
    </Card>
  </PaddedCard>
));

const List = ({
  recipes,
}: { recipes: Recipes}) => (
  <div>
    <Headline>Recipes</Headline>
    <CardDeck>
      { recipesList({ recipes }) }
      <PaddedCard>
        <Card width="15rem">
          <CardBlock>
            <Button color="success" href="/recipe/new">
              Create a new recipe
            </Button>
          </CardBlock>
        </Card>
      </PaddedCard>
    </CardDeck>
  </div>
);

List.propTypes = {
  recipes: PropTypes.shape({
    [PropTypes.string]: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    }),
  }),
};

List.defaultProps = {
  recipes: {},
};

export default List;
