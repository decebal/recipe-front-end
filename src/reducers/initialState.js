export default {
  recipes: [
    {
      id: 1,
      name: 'Keto Breakfast Tacos With Bacon and Guacamole',
      description: `Nutritional Information (Per Serving):
Calories: 387 Protein: 11g Carbs: 9g Net Carbs: 4g Sugar: 0g Sugar Alcohol:0g
Fiber: 5g
Fat: 35g
Saturated Fat: 16g Polyunsaturated: 3g Monounsaturated: 10g Trans fat: 0g
Cholesterol: 210g Sodium: 525mg Potassium: 369mg Vitamin A: 71%
Vitamin C: 11% Calcium: 2% Iron: 9%`,
      ingredients: [
        '1 tablespoon Brain Octane Oil',
        '2 pasture-raised eggs',
        '1 tablespoon grass-fed ghee',
        '1 medium organic avocado',
        '1/4 teaspoon Himalayan pink salt',
        '1/4 cup chopped organic romaine lettuce',
        '2 slices cooked uncured all natural bacon',
        '3 tablespoons diced cooked organic sweet potatoes',
        'Optional: garnish of organic micro cilantro',
      ],
      instructions: [
        'Heat a small skillet to medium heat and add 1 tablespoon of ghee.',
        'Crack one egg into the center of the skillet and pierce the yolk.',
        'Cook the egg for approximately 1-2 minutes on each side or until solid\n'
        + 'but not overcooked. Remove from pan and place on a paper towel or\n'
        + 'parchment paper-lined plate.',
        'Repeat with other egg. These will be your taco shells.',
        'In a small bowl, mash avocado along with 1 tablespoon of Brain Octane\n'
        + 'Oil plus Himalayan pink salt.',
      ],
    },
    {
      id: 2,
      name: 'Bulletproof Coffee Egg Latte',
      description: 'Nutritional Information (Per Serving):\n'
      + 'Calories: 331 Protein: 24g Carbs: 1g\n'
      + 'Fiber: 0g Sugar:0g Fat: 25g\n'
      + 'Saturated Fat: 15g Polyunsaturated: 2g Monounsaturated: 4g\n'
      + 'Trans fat: 0g Cholesterol: 402mg Sodium: 280mg\n'
      + 'Potassium: 138mg Vitamin A: 21mg Vitamin C: 0mg\n'
      + 'Calcium: 6mg Iron: 10mg',
      ingredients: [
        '8 ounces black coffee',
        '1 tablespoon\n'
        + 'grass-fed butter',
        '1 teaspoon Brain\n'
        + 'Octane Oil',
        '2 pasture-raised eggs',
        '1 scoop Vanilla\n'
        + 'Collagen Protein',
        '1/4 teaspoon of\n'
        + 'cinnamon',
      ],
      instructions: [
        'Add eggs, butter, oil and cinnamon to\n'
        + 'the blender.',
        'Add coffee and blend for 45 seconds\n'
        + 'on high.',
        'Add collagen protein and blend for\n'
        + '5 seconds on low.',
        'Top with cinnamon.',
      ],
    },
  ],
};
