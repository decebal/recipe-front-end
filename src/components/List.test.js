import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';
import initialState from '../reducers/initialState';


test('List renders correctly', () => {
  const tree = renderer
    .create(<List recipes={initialState.recipes} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
