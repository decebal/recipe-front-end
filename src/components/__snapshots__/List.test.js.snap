// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <h1
    className="sc-bdVaJa qBDhK"
  >
    Recipes
  </h1>
  <div
    className="CardDeck-hiovo2-0 fKfSga card-deck"
  >
    <div
      className="sc-bxivhb gJleJc"
    >
      <div
        className="Card-sc-18qyd5o-0 fVoFvn card"
      >
        <div
          className="card-block"
        >
          <a
            className="Button-h8watj-0 OlpfH btn btn-success"
            href="/recipe/new"
            onClick={[Function]}
          >
            Create a new recipe
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
